src/metabase/query_processor/middleware/wrap_value_literals.clj

```diff
@@ -43,12 +49,27 @@
      (when (types/temporal-field? field-info)
        {:unit :default}))))
 
+(defn- str-id-field->type-info
+  "Return _type info_ for a field with `field-name`, presumably from source query model.
+
+  Idea is, that if field has a string id, it may come from a model. If it _comes_ from a model, the model has to be
+  a source query. Hence the query with source model contains `:source-metadata` those could be mapped to `field-name`.
+
+  Returns `nil` when no matching field is found in `source-metadatas`."
+  [field-name source-metadatas]
+  (when (string? field-name)
+    (some (fn [{:keys [name] :as source-metadata}]
+            (when (= name field-name)
+              (select-keys source-metadata [:base_type :effective_type :database_type])))
```

Comment:
Shadowing functions like `name` can lead to a lot of head scratching.
```suggestion
    (some (fn [source-metadata]
            (when (= (:name source-metadata) field-name)
              (select-keys source-metadata [:base_type :effective_type :database_type])))
```

Reply:
Thanks.